### ISSUE TITLE
(Step 3) Search the mesh better than just brute force

### ISSUE THREAD

**Issue by zhangchuck** [zhangchuck/cme323risk#3]

**Comment by zhangchuck** [comment]
Could be use for mesh search for hyperparameter tuning.

**Comment by elpazn** [comment]
1. "Gradient descent" of weights on a single machine = batch-parallel /. At the end, have a batch reduce to find minimum

2. "Smart descent" by communicating with each other the results for each weight set -> batch-reduce (because multiple machines work on one pagerank).
